### DISCUSSION TITLE
Is there a way to process only on success and peek on error?

### DISCUSSION THREAD

**Discussion by EmmanuelMess** [foonathan/lexy#165] [Q&A] [answered]
I want to write code that "autobacktracks" whenever an error occurs, it seems that 
```
static constexpr auto rule = [] {
    auto ws = dsl::whitespace(dsl::ascii::space);
    auto condition = dsl::peek( rule using ws );
    
    return condition >> rule not using ws;
}();
```
is the only way. But this makes me write the same rule twice, but for the first one I *have* to use whitespace.

If I don't use this syntax I run the risk of a series of tokens being read by the wrong production and breaking unexpectedly without recovery.

(Also clangd seems annoyed that ws is two letters)

**Comment by foonathan** [comment, marked as answer]
This isn't how lexy is meant to be used, so there is no easy way to get it.
